src/NuGet.Status/Helpers/StorageService.cs

```diff
@@ -1,52 +1,131 @@
         {
-            var storageAccount = GetCloudStorageAccount();
+            if (_tableClient is null || _useManagedIdentity is false)
+            {
+                TableServiceClient tableServiceClient = GetTableServiceClient();
+                _tableClient = tableServiceClient.GetTableClient(MvcApplication.StatusConfiguration.TableName);
+            }
 
-            return storageAccount
-                .CreateCloudBlobClient()
-                .GetContainerReference(MvcApplication.StatusConfiguration.ContainerName);
+            return _tableClient;
         }
 
-        public CloudTable GetCloudTable()
+        private BlobContainerClient GetBlobContainerClient()
         {
-            var storageAccount = GetCloudStorageAccount();
+            BlobServiceClient blobServiceClient = GetBlobServiceClient();
+            return blobServiceClient.GetBlobContainerClient(MvcApplication.StatusConfiguration.ContainerName);
+        }
+
+        private BlobServiceClient GetBlobServiceClient()
+        {
+            if (_useManagedIdentity)
+            {
+                if (string.IsNullOrWhiteSpace(_managedIdentityClientId))
+                {
+                    // 1. Using MSI with DefaultAzureCredential
+                    return new BlobServiceClient(GetBlobServiceEndpoint(), new DefaultAzureCredential());
+                }
+                else
+                {
+                    // 2. Using MSI with ClientId
+                    return new BlobServiceClient(GetBlobServiceEndpoint(), new ManagedIdentityCredential(_managedIdentityClientId));
+                }
+            }
 
-            return storageAccount
-                .CreateCloudTableClient()
-                .GetTableReference(MvcApplication.StatusConfiguration.TableName);
+            // 3. Using SAS token
+            return new BlobServiceClient(_getStorageConnectionString());
         }
 
-        public CloudStorageAccount GetCloudStorageAccount()
+        private TableServiceClient GetTableServiceClient()
         {
-            return CloudStorageAccount.Parse(_getConnectionString());
+            if (_useManagedIdentity)
+            {
+                if (string.IsNullOrWhiteSpace(_managedIdentityClientId))
+                {
+                    // 1. Using MSI with DefaultAzureCredential
+                    return new TableServiceClient(GetTableServiceEndpoint(), new DefaultAzureCredential());
+                }
+                else
+                {
+                    // 2. Using MSI with ClientId
+                    return new TableServiceClient(GetTableServiceEndpoint(), new ManagedIdentityCredential(_managedIdentityClientId));
+                }
+            }
+
+            // 3. Using SAS token
+            return new TableServiceClient(_getStorageConnectionString());
+        }
+
+        private Uri GetBlobServiceEndpoint()
+        {
+            var temp = new BlobServiceClient(_getStorageConnectionString());
```

Comment:
Please add a comment regarding why storage client and table client are constructed differently. Or just make them work the same way :)

Reply:
Done. I've added a comment explaining why we do things differently for the Blob and Table endpoints